common/parser.hpp

```diff
@@ -194,50 +197,37 @@ struct commands_xml
     std::map<std::string, custom_formatter> custom_formatters;
 };
 
-inline std::string hexify(unsigned char n)
+// While working from files, convert the file content into recognized raw format
+inline void preprocess_data_payload(std::vector<std::string>& params)
 {
-    std::string res;
+    if (!params.size())
+        return;
 
-    do
-    {
-        res += "0123456789ABCDEF"[n % 16];
-        n >>= 4;
-    } while (n);
+    //The last parameter provided by the user can represent a fully-qualified file path
+    std::vector<std::string> modified_params = params;
 
-    std::reverse(res.begin(), res.end());
+    std::string str_param = params.back();
 
-    if (res.size() == 1)
+    if (regex_match(str_param, std::regex(".*\\.*$")))
```

Comment:
Why with regex?
What about `/` separator in Linux?
What if the filename has no dir separator?
Why not just assume it's a file and just check if the file is good?